pol-core/pol/syshookscript.cpp

```diff
@@ -604,5 +466,58 @@ size_t ExportScript::estimateSize() const
 {
   return sd.estimatedSize() + uoexec.sizeEstimate();
 }
+
+bool ExportScript::check_bool()
```

Comment:
I'm not sure if "check" is the best name for the methods, because it implies you're only validating the result but you actually read it. Did you consider other names?

Maybe an "expect_bool", "expect_int", etc could imply you read the result from the script with some expectation about the type